accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
+  - Import Kiwi Image
+
+- Implement a tool that handles the use cases above
+
+
+HWTYPE configurations
+---------------------
+
+This is currently implemented as a formula assigned to a group with special name.
+
+FIXME: pick one alternative
+
+Alternative 1:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+This is relatively small change to the code.
+
+Alternative 2:
+Central creation of groups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+YAML configuration
+------------------
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from YAML file.
+[Hub Users RFC](00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+Combined SUMA and Branch server
+-------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined SUMA and Branch server, these formulas should be applied on SUMA server itself.
+
+Terminals connect to SUMA directly, SUMA Proxy is not needed.
+
+
+FIXME: pick one alternative
+
+Alternative 1: Hub as a Salt Master for Combined SUMA and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries and disconnected operation
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and 
+  Combined Peripheral Servers will have non-optimal configuration
+- does not cover the "Single server" setup
+
+
+Alternative 2: Multi-master setup of Combined SUMA and Branch server
+
+- Salt minion running on Combined SUMA and Branch server has 2 masters - Hub and itself
```

Comment:
Branch server consist of several services which are configured via Formulas with Forms (via UI or API).
If we merge Branch server with Peripheral server, the formulas have to be applied on the Peripheral server itself - so the Peripheral server minion must be connected to Peripheral server master. At the same time, the minion must connect to the Hub master for Hub functionality.
According to Salt documentation, this should work as long as there are no state conflicts.

Reply:
So, If I understand correctly, a suse manager server can live on the same machine with the Branch server?
A peripheral server is just a suse manager server.
Hub server is a suse manager server with some more services runing.
Instead of register the peripheral server to itself, shouldn't we register it to the suse manager hub server, and configure the needed branch formulas from hub?

Reply:
> Instead of register the peripheral server to itself, shouldn't we register it to the suse manager hub server, and configure the needed branch formulas from hub?

Than this would be scenario 1 I think.

Reply:
Registering a SUSE Manager onto itself (or having a salt-minion attached to a salt-master on the same machine, for that matter) opens up a considerable can of worms and we have to have strong reasons to dig into that direction.

Among other things, consider that while salt-master is by and large an unprivileged service, salt-minion runs as root, so any user accessing the machine can potentially use the salt-master and get root access to the machine itself as well. Arguably that's not a whole lot better than accessing all minions in any case, but still it's not a nice proposition from a security perspective.

Reply:
I think this is not a problem for Retail.
Peripheral server will be managed by Hub admins who can access everything 
and optionally by local admins who should access for example all machines in one building.
There is no need for other roles.